extensions/amp-brid-player/0.1/amp-brid-player.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getLengthNumeral, isLayoutSizeDefined} from '../../../src/layout';
+import {loadPromise} from '../../../src/event-helper';
+import {setStyles} from '../../../src/style';
+
+class AmpBridPlayer extends AMP.BaseElement {
+
+  /** @override */
+  preconnectCallback(onLayout) {
+    this.preconnect.url('https://services.brid.tv', onLayout);
+    this.preconnect.url('https://cdn.brid.tv', onLayout);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    const width = this.element.getAttribute('width');
+    const height = this.element.getAttribute('height');
+
+    /** @private @const {number} */
+    this.width_ = getLengthNumeral(width);
+
+    /** @private @const {number} */
+    this.height_ = getLengthNumeral(height);
+
+    /** @private @const {string} */
+    this.partnerID_ = AMP.assert(
+        this.element.getAttribute('data-partner'),
+        'The data-partner attribute is required for <amp-brid-player> %s',
+        this.element);
+
+    /** @private @const {string} */
+    this.feedID_ = AMP.assert(
+        (this.element.getAttribute('data-video') ||
+        this.element.getAttribute('data-playlist')),
+        'Either the data-video or the data-playlist ' +
+        'attributes must be specified for <amp-brid-player> %s',
+        this.element);
+
+    if (!this.getPlaceholder()) {
+      this.buildImagePlaceholder_();
+    }
+  }
+
+  /** @override */
+  layoutCallback() {
+    const playerID = AMP.assert(this.element.getAttribute('data-player'),
+        'The data-player attribute is required for <amp-brid-player> %s',
+        this.element);
+
+    const partnerID = AMP.assert(
```

Comment:
I see. No problem then - tests sometimes have slightly different requirements. We will be addressing that soon.